modules/tinydnsbackend/tinydnsbackend.cc

```diff
@@ -164,6 +164,8 @@ void TinyDNSBackend::getAllDomains(vector<DomainInfo> *domains, bool include_dis
   d_cdbReader->searchAll();
   DNSResourceRecord rr;
 
+  std::unordered_set<DomainInfo> result;
+
   while (get(rr)) {
     if (rr.qtype.getCode() == QType::SOA) {
```

Comment:
You can replace the entire pull by this
 ```
std::unordered_set<DNSName> dupcheck;

 while (get(rr)) {
    if (rr.qtype.getCode() == QType::SOA && dupcheck.insert(rr.qname).second) {
```

Reply:
i didn't want to use dupcheck, because I feel we should be using unordered_set for the getAllDomains instead of Vector.

Reply:
but i didn't feel like doing it for getAllDomains now.